core/src/main/java/com/novoda/noplayer/exoplayer/mediasource/RendererTrackIndexExtractor.java

```diff
@@ -0,0 +1,21 @@
+package com.novoda.noplayer.exoplayer.mediasource;
+
+import com.google.android.exoplayer2.C;
+import com.novoda.noplayer.exoplayer.RendererTypeRequester;
+
+class RendererTrackIndexExtractor {
+
+    public int get(TrackType trackType, int mappedTrackInfoLength, RendererTypeRequester rendererTypeRequester) {
+        for (int i = 0; i < mappedTrackInfoLength; i++) {
+            int rendererType = rendererTypeRequester.getRendererTypeFor(i);
+
+            if ((trackType == TrackType.AUDIO && rendererType == C.TRACK_TYPE_AUDIO)
+                    || (trackType == TrackType.VIDEO && rendererType == C.TRACK_TYPE_VIDEO)
+                    || (trackType == TrackType.TEXT && rendererType == C.TRACK_TYPE_TEXT)) {
+                return i;
+            }
+        }
+
+        return -1;
```

Comment:
`But that would be really weird because you have to pass a track that you get from the player itself...` Maybe this means we should crash here ? This would be a bug/developper error right ?